### PR TITLE
gcc: update to 13.2.0.

### PR DESCRIPTION
**Pull request by oreo639**
[ci skip]
#### Testing the changes
- I tested the changes in this PR: **briefly**

You can test the ISOs here: https://drive.google.com/drive/u/0/folders/1y5dkzI9R60eDjRA4mVYBdnn9e9PDGwNt

### CODE REVIEW

**Comment by leahneukirchen** [comment]
We should use some gmp mirror, the devs probably block GitHub.

File: srcpkgs/gcc/template [hunk 1]
```diff
@@ -3,13 +3,13 @@
...
+revision=1
 _minorver="${version%.*}"
 _majorver="${_minorver%.*}"
-_gmp_version=6.2.1
-_mpfr_version=4.1.0
-_mpc_version=1.2.1
+_gmp_version=6.3.0
+_mpfr_version=4.2.0
+_mpc_version=1.3.1
 _isl_version=0.24
```
Comment (tornaria) [inline srcpkgs/gcc/template:13]: Maybe also update mpfr to 4.2.1 (#45725) and isl to 0.26 (#45951)?

**Comment by tornaria** [comment]
> We should use some gmp mirror, the devs probably block GitHub.

This works for the main gmp template: `${GNU_SITE}/gmp/gmp-${version}.tar.xz`

File: srcpkgs/gcc/template (hunk 1 above)
Comment (oreo639) [reply srcpkgs/gcc/template:13]: isl 0.25+ requires c++17 whereas gcc compiles with c++11.

Comment (leahneukirchen) [reply srcpkgs/gcc/template:13]: Is anyone really using gcc<7?

Comment (oreo639) [reply srcpkgs/gcc/template:13]: The issue is that gcc uses -std=c++11 and isl gets compiled through gcc's makefile so the flag gets passed on.

**Comment by metaleap** [comment]
Uh... "voting" for this one won't do jack, I guess... :grin:

**Comment by oreo639** [comment]
It is waiting on the glibc PR.

File: srcpkgs/gcc/template [hunk 2]
```diff
@@ -19,15 +19,15 @@ license="GFDL-1.2-or-later, GPL-3.0-or-later, LGPL-2.1-or-later"
 # *-musl builders have issues fetching https://ftp.gnu.org/gnu/mpc/mpc-1.1.0.tar.gz
 distfiles="
  ${GNU_SITE}/gcc/gcc-${version}/gcc-${version}.tar.xz
- https://gmplib.org/download/gmp/gmp-${_gmp_version}.tar.xz
+ ${GNU_SITE}/download/gmp/gmp-${_gmp_version}.tar.xz
```
Comment (averyterrel) [inline srcpkgs/gcc/template:22]: `${GNU_SITE}/download` should just be `${GNU_SITE}`

Comment (oreo639) [reply srcpkgs/gcc/template]: Thanks, I didn't notice sine the old source was still in my hostdir.

**Comment by simvux** [comment]
Out of curiosity, what is the process for updating a core package like this? 

It seems that we're bumping the revision for packages to force them to re-build with the new gcc/glibc version. However; how do we know which packages should be rebuilt and which shouldn't? 
Or are we using some sort of script that automatically bumps the revision for anything that depends on a specified package? 

After merge, will there still be packages that are built against the old glibc version but at runtime will link against the new installed glibc version?

**Comment by oreo639** [comment]
> It seems that we're bumping the revision for packages to force them to re-build with the new gcc/glibc version. However; how do we know which packages should be rebuilt and which shouldn't?

The libtool and mk-configure hard code gcc paths so they need to be rebuilt for a new gcc.

> After merge, will there still be packages that are built against the old glibc version but at runtime will link against the new installed glibc version?

Yes, that is by design.

**Comment by dataCobra** [comment]
Hey,

is there any way one can assist you with this?

I'm not familiar with gcc but maybe there are other ways to support you.

**Comment by 22h49** [comment]
Hi, I second the commenter above me

I'm also willing to help if there's any help needed, have experience with void templates.

**Review by leahneukirchen — APPROVED** [review]

**Comment by Calandracas606** [comment]
Trying to build cross-armv7l-linux-gnueabihf on x86_64-musl and getting a linker error.

```
/builddir/cross-armv7l-linux-gnueabihf-0.36/build_root/usr/bin/../lib/gcc/armv7l-linux-gnueabihf/13.2.0/../../../../armv7l-linux-gnueabihf/bin/ld: cannot find -lstdc++: No such file or directory
/builddir/cross-armv7l-linux-gnueabihf-0.36/build_root/usr/bin/../lib/gcc/armv7l-linux-gnueabihf/13.2.0/../../../../armv7l-linux-gnueabihf/bin/ld: cannot find -lgcc_s: No such file or directory
```

is a source bootstrap of masterdir needed?

**Comment by oreo639** [comment]
> Trying to build cross-armv7l-linux-gnueabihf on x86_64-musl and getting a linker error

cross compilers cannot be built in a masterdir where cross compilers are currently installed (xbps-src doesn't uninstall the cross compilers).

You should be able to just use
```
./xbps-src -m masterdir-cross binary-bootstrap
./xbps-src -m masterdir-cross pkg <pkg>
```

**Comment by oreo639** [comment]
> Hey,
> 
> is there any way one can assist you with this?

Sorry for the late response, feel free to report any issues you notice with the package.

I am running through compiling the all of the base ISO packages atm to test for regressions in the gcc 13 package. (I just noticed one relating to fortran and libquadmath)

**Comment by leahneukirchen** [comment]
:shipit:

**Comment by oreo639** [comment]
Merged.

**Pull request closed without merging**